src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx

```diff
@@ -1,23 +1,234 @@
+  const { t } = useTranslation();
+
+  const dispatch = useDispatch();
+  const reportData = useSelector((state: IRootState) => state.report.data);
+  const floor = reportData.floor;
+  const floorOptions = getOptions(cluster?.data.name || "").floor;
+
+  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value) {
+      dispatch(
+        updateAddReportData({
+          floor: e.target.value,
+        }),
+      );
+    }
+  };
+
+  return (
+    <>
+      <p className="basis-2/12 text-xs font-bold">
+        {t("addReport.content.select.setFloor.label", { ns: ["drawer"] })}
+      </p>
+      <Select
+        label={t("addReport.content.select.setFloor.label", {
+          ns: ["drawer"],
+        })}
+        selectedKeys={new Set([floor])}
+        onChange={handleSelectChange}
+      >
+        {floorOptions.map((s) => {
+          return (
+            <SelectItem key={s} value={s}>
+              {s ? s : "請選擇"}
+            </SelectItem>
+          );
+        })}
+      </Select>
+    </>
+  );
+};
+
+const TargetCategorySelect: React.FC<{
+  cluster: Cluster | null;
+}> = ({ cluster }) => {
+  const { t } = useTranslation();
+
+  const dispatch = useDispatch();
+  const reportData = useSelector((state: IRootState) => state.report.data);
+  const targetCategory = reportData.target.category;
+  const targetCategoryOptions =
+    cluster !== undefined && cluster !== null
+      ? getOptions(cluster.data.name).targetCategory[1]?.category || []
+      : [""]; // TODO: 要根據 floor 來決定值
+
+  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    if (e.target.value) {
+      dispatch(
+        updateAddReportData({
+          target: {
+            ...reportData.target,
+            category: e.target.value,
+          },
+        }),
+      );
+    }
+  };
+
+  return (
+    <>
+      <p className="basis-2/12 text-xs font-bold">
+        {t("addReport.content.select.setTargetCategory.label", {
+          ns: ["drawer"],
+        })}
+      </p>
+      <Select
+        label={t("addReport.content.select.setTargetCategory.label", {
+          ns: ["drawer"],
+        })}
+        selectedKeys={new Set([targetCategory])}
+        onChange={handleSelectChange}
+      >
+        {targetCategoryOptions.map((s) => {
+          return (
+            <SelectItem key={s} value={s}>
+              {s ? s : "請選擇"}
+            </SelectItem>
+          );
+        })}
+      </Select>
+    </>
+  );
+};
 
-const StatusSelect: React.FC = () => {
+const TargetNameSelect: React.FC<{ cluster: Cluster | null }> = ({
+  cluster,
+}) => {
   const { t } = useTranslation();
 
   const dispatch = useDispatch();
   const reportData = useSelector((state: IRootState) => state.report.data);
-  const status = reportData.status.type;
+  const targetName = reportData.target.name;
+  const targetNameOptions =
+    cluster !== undefined && cluster !== null
```

Comment:
If you already handle undefined by skeleton. It seems like in each Options, don't need to handle undefined